packages/editor/src/store/actions.js

```diff
@@ -196,22 +196,45 @@ export const savePost =
 			content,
 		};
 		dispatch( { type: 'REQUEST_POST_UPDATE_START', options } );
-		await registry
-			.dispatch( coreStore )
-			.saveEntityRecord(
-				'postType',
-				previousRecord.type,
-				edits,
+
+		let error = false;
+		try {
+			error = await applyFilters(
+				'editor.__unstablePreSavePost',
+				Promise.resolve( false ),
```

Comment:
If I'm not wrong we use this technique (a promise filter) because we don't support async actions yet right? Can we implement async action instead? `await doAsyncAction( 'editor.__unstablePreSavePost' )` 

I have another use-case for this and maybe it's time to fix that once and for all.

Reply:
Good idea. I'm working on this and one little obstacle is that `currentFilter()`/`currentAction()` don't make sense any more, because there can be multiple hooks running at the same time.

If you use the promise filter technique with sync hooks this problem is invisible. Because the hook merely sets up a promise chain and returns. Only later the actual data will flow through the chain. This will change when `runHook` will start waiting for each promise returned from a handler to resolve.

Reply:
This is a draft PR for async hooks: #64204